macast/server.py

```diff
@@ -101,15 +101,25 @@ def __init__(self, renderer, protocol, tool=None):
         self.tool_plugin = ToolPlugin(29, tool)
         self.tool_plugin.subscribe()
         check_ip_freq = Setting.get(SettingProperty.Check_IP_Frequency, 5)
-        self.ip_monitor = Monitor(cherrypy.engine, self.update_ip, check_ip_freq, name="IP_MONITOR_THREAD")
-        self.ip_monitor.subscribe()
+
+        if check_ip_freq:
+            self.ip_monitor = Monitor(cherrypy.engine, self.update_ip, check_ip_freq, name="IP_MONITOR_THREAD")
+            self.ip_monitor.subscribe()
+			
```

Comment:
It seems that there is no need for special judgment. If the frequency is less than or equal to 0, CherryPy's Monitor will not start.

Reply:
Indeed, I didn't look at cherrypy's source code before, but just read some of the documentation. However, the cost of adding judgment here should be lower than in the process of the Monitor instantiation.